controllers/flowlogspipeline/flp_common_objects.go

```diff
@@ -415,8 +427,9 @@ func (b *builder) addTransformStages(stage *config.PipelineBuilderStage) (*corev
 				},
 			},
 			TCPFlags: api.ConnTrackTCPFlags{
-				FieldName:           "Flags",
-				DetectEndConnection: true,
+				FieldName: "Flags",
+				//TODO: fix TCP detection that generates duplicated events
+				DetectEndConnection: false,
```

Comment:
@ronensc as discussed, this fix the duplicates when using direction. That seems to confirm your out of order theory :thinking: